app/scripts/directives/lifeCycleHook.js

```diff
@@ -0,0 +1,113 @@
+          }
+          return istag;
+        };
+
+        var setParam = function(path, defaultValue) {
+          _.set($scope.hookParams, path, _.get($scope.hookParams, path, defaultValue));
+        };
+        
+        var setHookParams = function() {
+          setParam(['failurePolicy'], "Abort");
+          if ($scope.actionType === "execNewPod") {
+            if (_.has($scope.removedHookParams, 'execNewPod')) {
+              $scope.hookParams.execNewPod = $scope.removedHookParams.execNewPod;
+              return;
+            }
+            setParam(['execNewPod', 'command'], []);
+            setParam(['execNewPod', 'env'], []);
+            setParam(['execNewPod', 'volumes'], []);
+            setParam(['execNewPod', 'containerName'], $scope.availableContainers[0] || "");
+          } else {
+            if (_.has($scope.removedHookParams, 'tagImages')) {
+              $scope.hookParams.tagImages = $scope.removedHookParams.tagImages;
+              return;
+            }
+            setParam(['tagImages', '0', 'containerName'], $scope.availableContainers[0] || "");
+            setParam(['tagImages', '0', 'to'], {});
+            $scope.istagHook = setImageOptions(_.head($scope.hookParams.tagImages).to);
+          }
+        };
+
+        setHookParams();
+
+        $scope.addHook = function() {
+          if (!_.isEmpty($scope.removedHookParams)) {
+            $scope.hookParams = $scope.removedHookParams;
+            $scope.view.hookExistes = true;
+            return;
+          }
+          $scope.hookParams = {};
+          setHookParams();
+          $scope.view.hookExistes = true;
+        };
+
+        $scope.removeHook = function() {
+          $scope.removedHookParams = $scope.hookParams;
+          delete $scope.hookParams;
+          $scope.view.hookExistes = false;
+          $scope.editForm.$setDirty();
+        };
+
+        $scope.actionTypeChange = function(actionType) {
+          $scope.actionType = actionType;
+          if (actionType === 'execNewPod') {
+
+            $scope.removedHookParams['tagImages'] = $scope.hookParams.tagImages;
+            delete $scope.hookParams.tagImages;
+          } else {
+            $scope.removedHookParams['execNewPod'] = $scope.hookParams.execNewPod;
+            delete $scope.hookParams.execNewPod;
+          }
+          setHookParams();
+          console.log($scope.hookParams);
+        };
+
+        $scope.$watch("istagHook.tagObject.tag", function() {
+          if (!_.has($scope.istagHook, ['tagObject', 'tag'])) {
+            return;
+          }
+          // Assamble image name when tag changes
+          _.set($scope.hookParams, ['tagImages', '0', 'to', 'kind'], 'ImageStreamTag');
+          _.set($scope.hookParams, ['tagImages', '0', 'to', 'namespace'], $scope.istagHook.namespace);
+          _.set($scope.hookParams, ['tagImages', '0', 'to', 'name'], $scope.istagHook.imageStream + ':' + $scope.istagHook.tagObject.tag);
```

Comment:
OK, but let's make sure we don't drop any others in the array. It might be better to show the values and make it read-only if there is more than one.